### ISSUE TITLE
1.21 service grafana create issue

### ISSUE THREAD

**Issue by NissesSenap** [grafana-operator/grafana-operator-experimental#1]
Hi @pb82 tried using the operator and I get some issues when creating a service:

```
2022-03-05T16:59:23.371+0100    ERROR   controller.grafana      reconciler error in stage   
    {"reconciler group": "grafana.integreatly.org", "reconciler kind": "Grafana",
     "name": "grafana", "namespace": "grafana", "stage": "service", 
     "error": "Service \"grafana-service\" is invalid: [spec.clusterIPs[0]: Invalid value: []string(nil): 
     primary clusterIP can not be unset, spec.ipFamilies[0]: Invalid value: []core.IPFamily(nil): primary ipFamily can not be unset]"}
```

How to recreate:

```shell
cat <<EOF | kind create cluster --image=kindest/node:v1.21.2 --config=-
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
  kubeadmConfigPatches:
  - |
    kind: InitConfiguration
    nodeRegistration:
      kubeletExtraArgs:
        node-labels: "ingress-ready=true"
  extraPortMappings:
  - containerPort: 80
    hostPort: 80
    protocol: TCP
  - containerPort: 443
    hostPort: 443
    protocol: TCP
EOF
```

I took a quick look and it seems like this was fixed in 1.22
https://github.com/kubernetes/kubernetes/issues/91459
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.22.md#api-change-2

It's not impossible that we will release this version after the 1.24 is released thus making 1.21 not supported. But at the same time we know that there will always be stranglers and it would be nice to support 1.21.

My grafana.yaml looks like this:

```
apiVersion: grafana.integreatly.org/v1beta1
kind: Grafana
metadata:
  name: grafana
  labels:
    app: grafana
spec:
  ingress:
    enabled: false
  config:
    log:
      mode: "console"
      level: "error"
    log.frontend:
      enabled: true
    auth:
      disable_login_form: False
      disable_signout_menu: True
    auth.anonymous:
      enabled: True
  resources:
    # Optionally specify container resources
    limits:
      cpu: 200m
      memory: 200Mi
    requests:
      cpu: 100m
      memory: 100Mi
```

**Comment by NissesSenap** [comment]
@pb82 didn't you fix this?

**Comment by weisdd** [comment]
@NissesSenap seems like the fix was added in `v1.21.5`:

_When using kubectl replace (or the equivalent API call) on a Service, the caller no longer needs to do a read-modify-write cycle to fetch the allocated values for .spec.clusterIP and .spec.ports[].nodePort. Instead the API server will automatically carry these forward from the original object when the new object does not specify them. (https://github.com/kubernetes/kubernetes/pull/104673, [@thockin](https://github.com/thockin)) [SIG Network]_
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.21.md#bug-or-regression-8

There's no kind with that tag, you could use `v1.21.10` instead. - Works just fine.
=> Let's close the issue :)

**Comment by NissesSenap** [comment]
Thanks for pointing this out. I will close the issue
